django/applications/catmaid/static/js/widgets/neuron_dendrogram.js

```diff
@@ -979,7 +979,10 @@
     var nodeName = function(showTags, showIds, showStrahler) {
       function addTag(d, wrapped) {
         if (d.tagged) {
-          return referenceTags + (wrapped.length > 0 ? " (" + wrapped + ")" : "");
+          var nodeTags = referenceTags.filter(function(t) {
+            return -1 !== tags[t].indexOf(d.id);
+          });
+          return nodeTags.join(",") + (wrapped.length > 0 ? " (" + wrapped + ")" : "");
```

Comment:
Would you mind moving the filter function definition to the scope in which `addTag` and `addStrahler` is defined? This way the function wouldn't be re-instantiated with every call to `addTag`. To still be able to use `d.id` you could provide it as context argument to `filter`, e.g.:

```
function isTaggedWith(t) {
 return -1 !== tags[t].indexOf(this);
}

[...]

var nodeTags = referenceTags.filter(isTaggedWith, d.id)
```